File: CS/CopyNodesExample/CopyNodesOperation.cs
using System;
using DevExpress.XtraTreeList;
using DevExpress.XtraTreeList.Nodes;
using DevExpress.XtraTreeList.Nodes.Operations;

namespace CopyNodesExample {
    public class CopyNodesOperation :TreeListOperation {
        TreeList DestTreeList;

        public CopyNodesOperation(TreeList destTreeList) {
            if (destTreeList == null)
                throw new ArgumentNullException("destTreeList");
            this.DestTreeList = destTreeList;
        }

        public override void Execute(TreeListNode node) {
            object[] values = new object[node.TreeList.Columns.Count];
            for (int i = 0; i < node.TreeList.Columns.Count; i++)
                values[i] = node.GetValue(i);
            if (node.ParentNode == null)
                DestTreeList.AppendNode(values, null);
            else DestTreeList.AppendNode(values, node.ParentNode.Id);
        }
    }
}

File: VB/CopyNodesExample/CopyNodesOperation.vb
Imports System
Imports DevExpress.XtraTreeList
Imports DevExpress.XtraTreeList.Nodes
Imports DevExpress.XtraTreeList.Nodes.Operations

Namespace CopyNodesExample

    Public Class CopyNodesOperation
        Inherits TreeListOperation

        Private DestTreeList As TreeList

        Public Sub New(ByVal destTreeList As TreeList)
            If destTreeList Is Nothing Then Throw New ArgumentNullException("destTreeList")
            Me.DestTreeList = destTreeList
        End Sub

        Public Overrides Sub Execute(ByVal node As TreeListNode)
            Dim values As Object() = New Object(node.TreeList.Columns.Count - 1) {}
            For i As Integer = 0 To node.TreeList.Columns.Count - 1
                values(i) = node.GetValue(i)
            Next

            If node.ParentNode Is Nothing Then
                DestTreeList.AppendNode(values, Nothing)
            Else
                DestTreeList.AppendNode(values, node.ParentNode.Id)
            End If
        End Sub
    End Class
End Namespace
